### PR TITLE
CM-910: How to Compare Two Lists Through One Property in C#

### PR DESCRIPTION
**Pull request by KonstantinosP7**
Please for your review

### CODE REVIEW

File: collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿using HowToCompareTwoListsThroughOneProperty.Models;
+
+namespace HowToCompareTwoListsThroughOneProperty.Methods;
+
+public class Methods
```
Comment (mahrncic) [inline collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs:5]: This whole class could be `static`. Also, let's give it a more meaningful name like `ListCompareMethods`

**Review by mahrncic — CHANGES_REQUESTED** [review]

File: collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                                   where orderList.Any(order => customer.Id == order.CustomerId)
+                                   select customer
+                                   ).ToList();
+
+        return customersWithOrders;
+    }
+
+    public static List<Customer> HashSetMethod(List<Customer> customerList, List<Order> orderList)
+    {
+        var customerIds = orderList.Select(i => i.CustomerId).Distinct().ToHashSet();
```
Comment (jshergal) [inline collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs:42]: We should remove `Distinct()`. Under the hood, it creates a `HashSet`. So this line is actually creating two `HashSets`

File: collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs (hunk 1 above)
Comment (KonstantinosP7) [reply collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs:5]: Changed to static and renamed to 'ListCompareMethods'.

File: collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs (hunk 2 above)
Comment (KonstantinosP7) [reply collections-lists/HowToCompareTwoListsThroughOneProperty/HowToCompareTwoListsThroughOneProperty/Methods/Methods.cs:42]: Removed 'Distinct()'

**Review by mahrncic — APPROVED** [review]

**Review by mahrncic — APPROVED** [review]

**Pull request merged**